frontend/src/pages/SignUp.jsx

```diff
@@ -2,12 +2,45 @@ import { Link } from "react-router-dom";
 import LogoRegular from "../components/icons/LogoRegular";
 import imageLoader from "../js/imageLoader";
 import { useState } from "react";
+import { PORT } from "../port";
 
 const SignUp = () => {
   const [imageIndex, setImageIndex] = useState(0);
+  const [isEmailDuplicated, setIsEmailDuplicated] = useState(false);
+  const [isUsernameDuplicated, setIsUsernameDuplicated] = useState(false);
 
   const handleSubmit = (e) => {
     e.preventDefault();
+    fetch(`http://localhost:${PORT}/auth/signup/`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        username: e.target.username.value,
+        email: e.target.email.value,
+        password: e.target.password.value,
+        firstName: e.target.firstName.value,
+        lastName: e.target.lastName.value,
+        country: e.target.country.value,
+        address: e.target.address.value,
+        zipcode: e.target.zipcode.value,
+        phone: e.target.phone.value,
+      }),
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        if (data?.error) throw data;
+        // Navigate to login page
+      })
```

Comment:
Would recommend moving this code  into a generic function and calling that function instead

Reply:
Done

Reply:
It's still not generic to handle any http method